test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,239 @@
+	_, err = exportClient.Cluster(e.exportClusterPath).Create(ctx, export, metav1.CreateOptions{})
+	require.NoError(t, err, "no error creating APIExport")
+
+	t.Logf("Retrying to create the APIExportEndpointSlice after the APIExport has been created")
+	require.Eventually(t, func() bool {
+		slice, err = sliceClient.Cluster(e.partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+		return err == nil
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected APIExportEndpointSlice creation to succeed")
+
+	require.Eventually(t, func() bool {
+		s, err := e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(s, apisv1alpha1.APIExportValid)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid APIExport")
+
+	t.Logf("Adding a Partition to the APIExportEndpointSlice")
+	patchedSlice := slice.DeepCopy()
+	patchedSlice.Spec.Partition = partition.Name
+	mergePatch, err := jsonpatch.CreateMergePatch(encodeJSON(t, slice), encodeJSON(t, patchedSlice))
+	require.NoError(t, err)
+
+	t.Logf("Try to patch the APIExportEndpointSlice to point at a Partition")
+	slice, err = e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Patch(ctx, slice.Name, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+	require.NoError(t, err, "error patching APIExportEndpointSlice")
+
+	require.Eventually(t, func() bool {
+		s, err := e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsFalse(s, apisv1alpha1.PartitionValid) && conditions.GetReason(s, apisv1alpha1.PartitionValid) == apisv1alpha1.PartitionInvalidReferenceReason
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected missing Partition")
+
+	t.Logf("Creating the missing Partition")
+	partitionClient := e.kcpClusterClient.TopologyV1alpha1().Partitions()
+	_, err = partitionClient.Cluster(e.partitionClusterPath).Create(ctx, partition, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating Partition")
+
+	require.Eventually(t, func() bool {
+		s, err := e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(s, apisv1alpha1.PartitionValid)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid Partition")
+
+	t.Logf("Checking that no endpoints have been populated")
+	require.Eventually(t, func() bool {
+		s, err := e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return len(s.Status.APIExportEndpoints) == 0
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "not expecting any endpoint")
+
```

Comment:
would it make sense to add a scenario with a partition that would select some shards ? could be just `1` in multi-shard env.

Reply:
As per the above this is covered with the private server.